### PR TITLE
chore(deps): Update Jest to v29

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

Bump:
  - jest to 29.1.2
  - @types/jest to 29.1.1
  - ts-jest to 29.0.3

Also included in this change are some Jest configuration changes.

The `transform` change resolves this deprecation warning:

```console
ts-jest[ts-jest-transformer] (WARN) Define `ts-jest` config under `globals` is deprecated. Please do
transform: {
    <transform_regex>: ['ts-jest', { /* ts-jest config goes here in Jest */ }],
},
```

The `snapshotFormat` allows us to retain the snapshot format from Jest 28 (we'll remove in a follow-up PR). See:
  - https://jestjs.io/docs/upgrading-to-jest29#snapshot-format
  - https://jestjs.io/blog/2022/08/25/jest-29

## How to test

See CI.

## How can we measure success?

Keeping up-to-date.

## Have we considered potential risks?

n/a

## Checklist

- [ ] I have listed any breaking changes, along with a migration path [^1]
- [ ] I have updated the documentation as required for the described changes [^2]

[^1]: Consider whether this is something that will mean changes to projects that have already been migrated, or to the CDK CLI tool. If changes are required, consider adding a checklist here and/or linking to related PRs.
[^2]: If you are adding a new construct or pattern, has new documentation been added? If you are amending defaults or changing behaviour, are the existing docs still valid?

### CODE REVIEW

**Comment by akash1810** [comment]
Replaces https://github.com/guardian/cdk/pull/1392.

File: src/bin/commands/new-project/utils/utils.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export function getCommands(
   return {
     installDependencies: () => runTask(packageManager, cwd, "install"),
     lint: () => runTask(packageManager, cwd, "run lint"),
-    test: () => runTask(packageManager, cwd, "test"),
+    test: () => runTask(packageManager, cwd, "test -- -u"),
```
Comment (akash1810) [inline src/bin/commands/new-project/utils/utils.ts:31]: See https://github.com/guardian/cdk/pull/1392/files#r925215363 and https://github.com/facebook/jest/pull/12378.

**Review by nicl — APPROVED** [review]
🎉

File: jest.config.js [hunk 2]
```diff
@@ -16,9 +21,10 @@ module.exports = {
     - #448
    */
   modulePathIgnorePatterns: ["<rootDir>/lib"],
-  globals: {
-    "ts-jest": {
-      tsconfig: "tsconfig.test.json",
-    },
+
+  // (temporary) Retain snapshot format from Jest 28. See https://jestjs.io/docs/upgrading-to-jest29#snapshot-format.
```
Comment (akash1810) [inline jest.config.js:25]: https://github.com/guardian/cdk/pull/1531

**Pull request merged**